bundles/binding/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/EcobeeBinding.java

```diff
@@ -238,27 +255,32 @@ protected String getName() {
 	 */
 	@Override
 	protected void execute() {
-		logger.trace("Querying Ecobee API");
-
 		try {
 			for (String userid : credentialsCache.keySet()) {
 				OAuthCredentials oauthCredentials = getOAuthCredentials(userid);
 
-				Selection selection = createSelection(oauthCredentials);
-				if (selection == null) {
-					logger.debug("Nothing to retrieve for '{}'; skipping thermostat retrieval.",
-							oauthCredentials.userid);
-					continue;
-				}
+				if (oauthCredentials.pollTimeExpired()) {
+					// schedule the next poll at the standard refresh interval
+					oauthCredentials.schedulePoll(this.refreshInterval);
 
-				if (oauthCredentials.noAccessToken()) {
-					if (!oauthCredentials.refreshTokens()) {
-						logger.warn("Periodic poll skipped for '{}'.", oauthCredentials.userid);
+					logger.trace("Querying Ecobee API for instance {}", oauthCredentials.userid);
+
+					Selection selection = createSelection(oauthCredentials);
+					if (selection == null) {
+						logger.debug("Nothing to retrieve for '{}'; skipping thermostat retrieval.",
+								oauthCredentials.userid);
 						continue;
 					}
-				}
 
-				readEcobee(oauthCredentials, selection);
+					if (oauthCredentials.noAccessToken()) {
+						if (!oauthCredentials.refreshTokens()) {
+							logger.warn("Periodic poll skipped for '{}'.", oauthCredentials.userid);
```

Comment:
it might be helpful to the reason why polling is skipped (due to a missing token)